scene/resources/mesh_library.cpp

```diff
@@ -255,12 +256,35 @@ int MeshLibrary::get_last_unused_item_id() const {
 }
 
 void MeshLibrary::_set_item_shapes(int p_item, const Array &p_shapes) {
-	ERR_FAIL_COND(p_shapes.size() & 1);
+	Array arr_shapes = p_shapes;
+	int size = p_shapes.size();
+	if (size & 1) {
+		ERR_FAIL_COND_MSG(!item_map.has(p_item), "Requested for nonexistent MeshLibrary item '" + itos(p_item) + "'.");
+		int prev_size = item_map[p_item].shapes.size() * 2;
+
+		if (prev_size < size) {
+			// Check if last element is a shape
+			Ref<Shape3D> shape = arr_shapes[size - 1];
+			if (shape.is_null()) {
+				Ref<BoxShape3D> box_shape;
+				box_shape.instantiate();
+				arr_shapes[size - 1] = box_shape;
```

Comment:
I'm not sure about this, that seems pretty arbitrary, and could lead to a lot of instantiations that will then need to be freed and replaced by what the user wants to use.

Wouldn't it be sufficient to just make the step 2 for the SpinBox?

Reply:
> Wouldn't it be sufficient to just make the step 2 for the SpinBox?

Well not sure that's actually possible currently in the array editor.

Reply:
The array is expecting to have a shape. If there is anything else, even an empty `Variant()` the array won't update and the size won't change. A few lines below that piece of code you can find
```gdscript
if (sd.shape.is_valid()) {
	shapes.push_back(sd);
}
```
So, if it isn't a valid shape, nothing happens. Could you think of other way of doing it? It is absolutely arbitrary but is the way I found to do it.